src/solver.f90

```diff
@@ -489,20 +535,86 @@ subroutine curl(self, o_i_hat, o_j_hat, o_k_hat, u, v, w)
 
    end subroutine curl
 
-   subroutine run(self, n_iter, u_out, v_out, w_out)
+   subroutine poisson_fft(self, pressure, div_u)
+      implicit none
+
+      class(solver_t) :: self
+      class(field_t), intent(inout) :: pressure
+      class(field_t), intent(in) :: div_u
+
+      ! call forward FFT
+      ! output array in spectral space is stored at poisson_fft class
+      call self%backend%poisson_fft%fft_forward(div_u)
+
+      ! postprocess
+      call self%backend%poisson_fft%fft_postprocess
+
+      ! call backward FFT
+      call self%backend%poisson_fft%fft_backward(pressure)
+
+   end subroutine poisson_fft
+
+   subroutine poisson_cg(self, pressure, div_u)
+      implicit none
+
+      class(solver_t) :: self
+      class(field_t), intent(inout) :: pressure
+      class(field_t), intent(in) :: div_u
+
```

Comment:
```suggestion
print*, "CG Poisson solver not yet implemented."
```

Reply:
Ah, I committed this already but had to take it back. Printing here would result in unnecessary output every iteration when doing a performance test. Now we clarify CG is not implemented up when we initiate the poisson solver